src/test/java/org/terasology/logic/health/RegenTest.java

```diff
@@ -99,4 +106,32 @@ public void zeroRegenTest() {
 
         assertFalse(player.hasComponent(RegenComponent.class));
     }
+
+    @Test
+    public void regenTest() {
+        HealthComponent healthComponent = new HealthComponent();
+        healthComponent.currentHealth = 100;
+        healthComponent.maxHealth = 100;
+        healthComponent.waitBeforeRegen = 1;
+        healthComponent.regenRate = 1;
+
+        final EntityRef player = entityManager.create();
+        player.addComponent(new PlayerCharacterComponent());
+        player.addComponent(healthComponent);
+
+        player.send(new DoDamageEvent(5));
+        helper.runWhile(200, () -> true);
+        assertEquals(healthComponent.currentHealth, 95);
+
+        // wait 1 second before regen starts
+        float regenStarts = time.getGameTime() + healthComponent.waitBeforeRegen + BUFFER;
+        helper.runWhile(() -> time.getGameTime() <= regenStarts);
```

Comment:
and this is like a `sleep` but in GameTime units. Not quite sure how to do that one.

Reply:
I thought the `runWhile`/`runUntil` helper methods are exactly for this purpose? :thinking: I think in other places I uses a different pattern:

```
helper.runWhile(healthComponent.waitBeforeRegen + BUFFER, () -> true)
```

This runs indefinitely, but at max until the timeout of `healthComponent.waitBeforeRegen + BUFFER` is reached (I think the timeout uses a `long` for ms  while the game time here is `float` in seconds).